src/client.ts

```diff
@@ -230,17 +230,28 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
 
     this.userPromise = new Promise(resolve => {
       this.userPromiseResolver = resolve;
-    }).then(() => {
-      this.isUserReady = true;
-      return { success: true };
+    }).then(isUserResolved => {
+      if (isUserResolved) {
+        this.isUserReady = true;
+        return { success: true };
+      }
+      return { success: false, message: 'Failed to fetch qualified segments.' };
```

Comment:
Same here. Can we get the reason of user resolve failure from the caller?  
There can be other reasons in future for failure.